## Milestone 1

Exercise 1) Now, try to create the final scores of 20 students in Jupyter
Notebook as follows:

In [22]:
import scipy
from scipy.cluster.vq import kmeans ,vq , whiten
# Score of 20 students in on subjects
student_scores = [[80] , [78] , [55] , [60] , [62] ,
[59] , [48] , [49] , [51] , [54] ,
[72] , [95] , [78] , [54] , [50] ,
[61] , [57] , [56] , [65] , [53]];

Exercise 2) Next, normalize the raw data and calculate the centroids for
classifying two group of students:

In [2]:
student_scores = whiten ( student_scores )
centroids,_  = kmeans ( student_scores ,2)

Exercise 3

In [3]:
result ,_ = vq( student_scores , centroids )
print('The result is',result)

The result is [0 0 1 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1]


It means data observation is nearest the centroid by 0 and 1. Zero will near the first centroid, and One is near the second centroid.

Exercise 4

In [4]:
centroids,_ = kmeans(student_scores, 4)
result ,_ = vq( student_scores , centroids )
print('The result is',result)

The result is [0 0 3 1 1 1 3 3 3 3 0 2 0 3 3 1 1 3 1 3]


Exercise 5

In [5]:
student_scores = [[80 , 72] , [78 , 56] , [55 , 64] , [60 , 61] , [62 , 45] ,
[59 , 71] , [48 , 85] , [49 , 45] , [51 , 55] , [54 , 62] ,
[72 , 81] , [95 , 81] , [78 , 92] , [54 , 80] , [50 , 50] ,
[61 , 65] , [57 , 62] , [56 , 55] , [65 , 63] , [53 , 72]];

In [6]:
student_scores = whiten(student_scores)
centroids,_ = kmeans(student_scores, 4)
result,_ = vq(student_scores, centroids)
print('The result is',result)

The result is [2 2 3 3 0 3 1 0 0 3 2 2 2 1 0 3 3 0 3 1]


It means like exercise 3 but have 4 clusters.

## Milestone 2

Exercise 6

In [7]:
import scipy.constants
print ("The PI is %.16f"% scipy.constants.pi)
print ("The speed of light is c = %.1F"% scipy.constants.c)
print ("The newton ’s gravity constant is G = %.1F"% scipy.constants.g)

The PI is 3.1415926535897931
The speed of light is c = 299792458.0
The newton ’s gravity constant is G = 9.8


Exercise 7

In [8]:
scipy.constants.find("gram")

['atomic mass unit-kilogram relationship',
 'electron volt-kilogram relationship',
 'hartree-kilogram relationship',
 'hertz-kilogram relationship',
 'inverse meter-kilogram relationship',
 'joule-kilogram relationship',
 'kelvin-kilogram relationship',
 'kilogram-atomic mass unit relationship',
 'kilogram-electron volt relationship',
 'kilogram-hartree relationship',
 'kilogram-hertz relationship',
 'kilogram-inverse meter relationship',
 'kilogram-joule relationship',
 'kilogram-kelvin relationship']

The find fuction is to find constants which contain "gram".

Exercise 8

In [9]:
print(scipy.constants.physical_constants["atomic mass unit-kilogram relationship"])

(1.6605390666e-27, 'kg', 5e-37)


Exercise 9

In [10]:
m = 57
c = scipy.constants.c
E = m*(c**2)
print('Energy is {}'.format(E))

Energy is 5.122904518799861e+18


## Milestone 3

Exercise 10

In [13]:
import scipy.stats as stats
import numpy as np
x = np.array([19 ,18 ,21 ,16 ,15 ,17 ,20 ,18]);
print(stats.tmin(x) ,stats.tmax(x) ,stats.tmean(x) ,stats.tvar(x));

15 21 18.0 4.0


Exercise 11

In [14]:
print(stats.describe(x));

DescribeResult(nobs=8, minmax=(15, 21), mean=18.0, variance=4.0, skewness=0.0, kurtosis=-1.0)


Exercise 12

In [19]:
import scipy.stats
import numpy as np
student_scores = [[80] , [78] , [55] , [60] , [62] ,
[59] , [48] , [49] , [51] , [54] ,
[72] , [95] , [78] , [54] , [50] ,
[61] , [57] , [56] , [65] , [53]];
print (stats.tmean(student_scores))
print (stats.tstd(student_scores))
print ((62 - stats.tmean(student_scores))/stats.tstd(student_scores))

61.85
[12.55838994]
[0.01194421]


61.85 is the mean of student scores.
12.55 is the standard deviation of student scores.
0.0119 is the z-score of 62.

Exercise 13

In [31]:
for i in student_scores:
    z = ((i - stats.tmean(student_scores))/stats.tstd(student_scores))
    print(z)

[1.44524896]
[1.28599288]
[-0.54545209]
[-0.14731188]
[0.01194421]
[-0.22693992]
[-1.10284838]
[-1.02322034]
[-0.86396425]
[-0.62508013]
[0.80822462]
[2.63966959]
[1.28599288]
[-0.62508013]
[-0.9435923]
[-0.06768384]
[-0.386196]
[-0.46582404]
[0.25082833]
[-0.70470817]
